x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -221,15 +396,31 @@ const getQueryMappingForComparator = (comparator: Comparator) => {
   return queryMappings[comparator];
 };
 
-const getResults = async (
-  query: object,
-  index: string,
-  callCluster: AlertServices['callCluster']
-) => {
-  return await callCluster('count', {
-    body: query,
-    index,
-  });
+const getUngroupedResults = async (query: object, callCluster: AlertServices['callCluster']) => {
+  return decodeOrThrow(UngroupedSearchQueryResponseRT)(await callCluster('search', query));
+};
+
+const getGroupedResults = async (query: any, callCluster: AlertServices['callCluster']) => {
```

Comment:
Any reason to define the `query` as `any`?

```suggestion
const getGroupedResults = async (query: object, callCluster: AlertServices['callCluster']) => {
```

Reply:
So, this was originally to appease typechecking when `queryWithAfterKey.body.aggregations.groups.composite.after = lastAfterKey;` is called. Instead I've changed`const queryWithAfterKey = { ...query };` to `const queryWithAfterKey: any = { ...query };`, so that the parameter can stay aligned with the ungrouped function.

Ultimately, I know `any` is never a good idea. But in this scenario where the ES responses are typechecked, and the queries are dependant on the alert params, I'm not sure building out a complex type for the query is needed? What's your opinion?

Reply:
ah, I see - that makes sense :+1: thanks for the explanation